### ISSUE TITLE
Attributes of HDF5 root group cannot be accessed

### ISSUE THREAD

**Issue by jgreitemann** [ALPSCore/ALPSCore#263]
Associating attributes with the root element of a HDF5 file does currently not work:

    alps::hdf5::archive ar(filename, "w");
    ar["/@key"] << value;

throws the following exception:
```
terminate called after throwing an instance of 'alps::hdf5::path_not_found'
  what():  unknown path: 
In /home/j/J.Greitemann/opensource/ALPSCore/hdf5/src/archive.cpp on 1020 in write
    /home/j/J.Greitemann/.local/lib/libalps-hdf5.so(_ZNK4alps4hdf57archive5writeENSt7__cxx11$
2basic_stringIcSt11char_traitsIcESaIcEEEd+0x1a05) [0x7f995e368d15]
    /home/j/J.Greitemann/.local/lib/libalps-hdf5.so(_ZN4alps4hdf54saveERNS0_7archiveERKNSt7_$
cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKdSt6vectorImSaImEESF_SF_+0x65) [0x7f995e3bbc$
5]
```

Doing the same for a subgroup works as expected. To my knowledge, the root group should also be able to hold attributes. At least using `h5py` it's possible.

**Comment by galexv** [comment]
It is a subset of issue #264. Working on it...

**Comment by galexv** [comment]
Fixed by commit c7c64ca.
